services/src/atdd/src/test/java/org/openkilda/atdd/TopologyEventsBasicTest.java

```diff
@@ -64,56 +82,57 @@ public void multiple_links_exist_between_all_switches() throws Exception {
     }
 
     @When("^a link is dropped in the middle$")
-    public void a_link_is_dropped_in_the_middle() throws Exception {
+    public void linkIsDroppedInTheMiddle() throws Exception {
         List<IslInfoData> links = LinksUtils.dumpLinks();
         IslInfoData middleLink = getMiddleLink(links);
 
         PathNode node = middleLink.getPath().get(0);
         assertTrue(LinksUtils.islFail(getSwitchName(node.getSwitchId()), String.valueOf(node.getPortNo())));
-    }
 
-    @Then("^the link will have no health checks$")
-    public void the_link_will_have_no_health_checks() throws Exception {
-        List<IslInfoData> links = LinksUtils.dumpLinks();
-        List<IslInfoData> cutLinks = links.stream()
-                .filter(isl -> isl.getState() != IslChangeType.DISCOVERED)
-                .collect(Collectors.toList());
-
-        assertFalse("Link should be cut", cutLinks.isEmpty());
-        assertThat("Only one link should be cut", cutLinks.size(), is(1));
+        manipulatedLinkId = middleLink.getId();
     }
 
-    @Then("^the link disappears from the topology engine\\.$")
-    public void the_link_disappears_from_the_topology_engine() throws Exception {
-        List<IslInfoData> links = LinksUtils.dumpLinks();
+    @Then("^the link disappears from the topology engine in (\\d+) seconds\\.$")
+    public void theLinkDisappearsFromTheTopologyEngine(int timeout) throws Exception {
+        List<String> cutLinks = Failsafe.with(retryPolicy
+                .retryIf(links -> links instanceof List && ((List) links).isEmpty()))
+                .get(() -> LinksUtils.dumpLinks().stream()
+                        .filter(isl -> isl.getState() != IslChangeType.DISCOVERED)
+                        .map(IslInfoData::getId)
+                        .collect(Collectors.toList())
+                );
```

Comment:
you should also verify that it took **not less** than `timeout` before link got cut. It also should not take too long. I think  it should be some kind of +- window verification as we did with traffic.